crates/uv-fs/src/lib.rs

```diff
@@ -664,29 +663,28 @@ impl LockedFile {
             "Checking lock for `{resource}` at `{}`",
             file.path().user_display()
         );
-        match file.file().try_lock_exclusive() {
+        match file.file().try_lock() {
             Ok(()) => {
                 debug!("Acquired lock for `{resource}`");
                 Ok(Self(file))
             }
             Err(err) => {
-                // Log error code and enum kind to help debugging more exotic failures.
-                if err.kind() != std::io::ErrorKind::WouldBlock {
+                if !matches!(err, std::fs::TryLockError::WouldBlock) {
+                    // Log error code and enum kind to help debugging more exotic failures.
                     debug!("Try lock error: {err:?}");
                 }
                 info!(
                     "Waiting to acquire lock for `{resource}` at `{}`",
                     file.path().user_display(),
                 );
-                file.file().lock_exclusive().map_err(|err| {
+                file.file().lock().map_err(|err| {
                     // Not an fs_err method, we need to build our own path context
```

Comment:
Once https://github.com/andrewhickman/fs-err/issues/74 is fixed, we can simplify this code by using fs_err methods directly which have the paths attached to the error message.